js/ui/marker.js

```diff
@@ -13,6 +13,10 @@ var Point = require('point-geometry');
  * @param {HTMLElement=} element DOM element to use as a marker (creates a div element by default)
  * @param {Object=} options
  * @param {PointLike=} options.offset The offset in pixels as a [`PointLike`](#PointLike) object to apply relative to the element's top left corner. Negatives indicate left and up.
+ * @param {boolean=} [options.pitchScale=false] Whether the marker should scale smaller into the background or larger into the foreground when the map is pitched and rotated
+ * @param {boolean=} [options.zoomScale=false] Whether the marker should scale with the map as you zoom in and out
+ * @param {number=} [options.nativeZoom=map.getZoom()] The native zoom level of the marker. If the map is at this zoom the marker will not be scaled. If zoom is greater the marker will be enlarged. If zoom is less the marker will be shrunk.
+ * @param {number=} [options.zoomFactor=1.6] The multiple by which to scale the size of the marker between each zoom level
```

Comment:
> But anything that falls outside that range is bounded by the highest and lowest defined stops

This is the intended behaviour of GL functions and reflects how GL styles are crafted.

 > How do I get it to extrapolate out the correct values beyond the first and last stops without having to rewrite the logic in the constructor?

If you want extrapolation over a larger domain, you need to declare a larger domain 😄 

```js
var f = glfunc.interpolated({
    stops: [[1, 1], [20, 5]],
    base: 2
});
```